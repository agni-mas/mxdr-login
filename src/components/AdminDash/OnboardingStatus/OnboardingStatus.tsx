import react, { FC, useState, useEffect } from 'react';
import GridItem from '../../layout/GridItem/GridItem';
import Button from '@mui/material/Button';
import PieChartComponent from '../../global/PieChart/PieChart';
import { Skeleton } from '@mui/material';
import useFetchSecurityData from '../../../hooks/useFetchSecurityData';


const styles = {

}

interface LCPAlertsProps { }

const LCPAlerts: FC<LCPAlertsProps> = () => {
    const [onboardingStatus, setOnboardingStatus] = useState([])

    const { data, loading } = useFetchSecurityData('onboardingStatus.json')

    useEffect(() => {
        fetchData()
    }, [loading, data])

    const fetchData = () => {
        try {
            let rawData = (data) ?? []
            setOnboardingStatus(rawData)
        } catch (err) {
            console.error(err)
        }
    }


    return (
        <GridItem title="Onboarding Status">
            {data ? (<PieChartComponent insideLabel="Total Onboarding" data={onboardingStatus} type='admin' />) : (<Skeleton variant="rectangular" animation="wave" height={200} />)}
            <Button color='secondary' variant="outlined" style={{ float: 'right', borderRadius: 28, }}>View Details</Button>
        </GridItem>
    )
}

export default LCPAlerts;