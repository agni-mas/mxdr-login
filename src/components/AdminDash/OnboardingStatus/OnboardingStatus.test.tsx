import React from 'react';
import { render } from "@testing-library/react";
import OnboardingStatus from './OnboardingStatus'


it("renders <OnboardingStatus />", () => {
    const { asFragment } = render(<OnboardingStatus /> );
    expect(asFragment()).toMatchSnapshot();
});
