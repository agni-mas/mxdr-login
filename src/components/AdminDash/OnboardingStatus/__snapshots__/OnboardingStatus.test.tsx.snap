// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <OnboardingStatus /> 1`] = `
<DocumentFragment>
  <section
    class="item"
  >
    <h2
      class="grid-item-title"
    >
      Onboarding Status
    </h2>
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
      style="height: 200px;"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
      style="float: right; border-radius: 28px;"
      tabindex="0"
      type="button"
    >
      View Details
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </section>
</DocumentFragment>
`;
